### ISSUE TITLE
cookie_expiration_period for aws_lb_cookie_stickiness_policy is not optional

### ISSUE THREAD

**Issue by ckelner** [hashicorp/terraform#3208]
Reference: https://www.terraform.io/docs/providers/aws/r/lb_cookie_stickiness_policy.html
> cookie_expiration_period - (Optional) The time period after which the session cookie should be considered stale, expressed in seconds.

However, if I leave this argument out of my resource definition, i.e.
```
resource "aws_lb_cookie_stickiness_policy" "foo" {
  name = "${var.project}-${var.env}-stickyness-policy"
  load_balancer = "${aws_elb.gor-frontend.id}"
  lb_port = "${var.elb_port}"
}
```

I get the following error from AWS/Terraform during my `apply`:
```
aws_lb_cookie_stickiness_policy.foo: Creating...
  lb_port:       "" => "8080"
  load_balancer: "" => "grid-gor-staging"
  name:          "" => "grid-gor-staging-stickyness-policy"
aws_lb_cookie_stickiness_policy.foo: Error: 1 error(s) occurred:

* Error creating LBCookieStickinessPolicy: ValidationError: Cookie expiration period should be a greater than 0.
	status code: 400, request id: [03ac88a5-57d4-11e5-9914-6defe129f2a1]
Error applying plan:

1 error(s) occurred:

* Error creating LBCookieStickinessPolicy: ValidationError: Cookie expiration period should be a greater than 0.
	status code: 400, request id: [03ac88a5-57d4-11e5-9914-6defe129f2a1]
```

I decided to do some digging.  I wound up here: [resource_aws_lb_cookie_stickiness_policy.go#L56](https://github.com/hashicorp/terraform/blob/b91df72371231e583d6a33b5ed8027d914def8ba/builtin/providers/aws/resource_aws_lb_cookie_stickiness_policy.go#L56)
```
aws.Int64(int64(d.Get("cookie_expiration_period").(int)))
```

I decided to exercise this logic with a "null" value in Go (note, I am not a Go dev, so apologies if this is ugly): [terraform-testing.go](https://github.com/ckelner/play-with-go/blob/fb49ba83b7485d1312c803862619544ce91d8773/terraform-testing.go)
```
var cookie_expire int
fmt.Println(int64(cookie_expire))
fmt.Println(aws.Int64(int64(cookie_expire)))
```

The output of `go run terraform-testing.go` looks like:
```
0
0xc20800a8b0
```

I assume there needs to be some sort of "null check" to prevent sending this argument to AWS.

**Comment by catsby** [comment]
Thanks @ckelner – your findings are correct. `d.Get` will return the default value for an attribute, here `0`. Zero is not a valid expiration period and should simply be omitted. I've patched this in #3908 if you're interested. 

Thanks!

**Comment by ckelner** [comment]
Thanks! Looks good :+1: appreciate you taking care of this
